05/ContactForm.js

```diff
@@ -1,10 +1,179 @@
-import React from 'react';
-
-import account from './account';
+/* eslint-disable no-console */
+import React, { useReducer, useRef } from 'react';
+import emailjs from '@emailjs/browser';
+import { SERVICE_ID, TEMPLATE_ID, PUBLIC_KEY } from './account';
 
 const ContactForm = () => {
-    console.log(account);
-    return <form>??</form>;
+    const form = useRef();
+
+    const init = {
+        firstName: '',
+        email: '',
+        phone: '',
+        subject: '',
+        textArea: '',
+        errors: {
+            firstName: '',
+            email: '',
+            subject: '',
+            textArea: '',
+        },
+    };
+    const isValidEmail = (email) => {
+        const emailRegex = /^[a-zA-Z0-9._%+-]+@[a-zA-Z0-9.-]+\.[a-zA-Z]{2,4}$/;
+        return emailRegex.test(email);
+    };
+    // eslint-disable-next-line no-unused-vars
+    const isValidPhone = (phone) => {
+        const phoneRegex = /^\d{9}$/;
+        return phoneRegex.test(phone);
+    };
+
+    const validateFields = (values) => {
```

Comment:
Należy utworzyć funkcję, która będzie sprawdzać tylko wskazane pole np. `validateField(name, value)`.
Wtedy będziesz mógł uruchamiać walidacje dla jednego (aktualnego pola)

Najlepiej mieć "uniwersalną" walidację, taką jaką zaproponowałem w tym artykule:
https://devmentor.pl/b/walidacja-formularza-w-javascript

To będziesz mógł napisać funkcję, która waliduje tylko wskazane pole, gdzie jest nią zazwyczaj funkcja wew. `.forEach`:
```
fields.forEach(function (field) {
    const { name, label, required, type, pattern } = field;
    // wykorzystuję właściwość obiektu o nazwie name, by pobrać wartość konkretnego elementu formularza
    const value = formEl.elements[name].value;
```